src/Raven.Server/Documents/Sharding/Handlers/ReshardingHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Threading.Tasks;
+using Raven.Client.Exceptions.Database;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Raven.Server.Web;
+
+namespace Raven.Server.Documents.Sharding.Handlers
+{
+    public class ReshardingHandler : ServerRequestHandler
+    {
+        [RavenAction("/admin/resharding", "GET", AuthorizationStatus.Operator)]
+        public async Task StartResharding()
+        {
+            var database = GetStringQueryString("database");
```

Comment:
Why isn't this a database request handler?

Reply:
Because database request handle require the actual database e.g. Northwind$0, a sharded database request handler require a normal handler first. 
To avoid all this I used the server request handler.

Reply:
Okay, make sense.